modules/openthread/platform/platform.h

```diff
@@ -0,0 +1,272 @@
+ *  CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ *  ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ *  POSSIBILITY OF SUCH DAMAGE.
+ */
+
+/**
+ * @file
+ *   This file includes the platform-specific initializers.
+ *
+ */
+
+#ifndef PLATFORM_H_
+#define PLATFORM_H_
+
+#include <stdint.h>
+#include <openthread/config.h>
+#include <openthread/instance.h>
+#include <openthread-system.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#define OT_PLAT_DBG_LVL 0
+
+#define OT_PLAT_DBG_LEVEL_NONE 0
+#define OT_PLAT_DBG_LEVEL_ERR 1
+#define OT_PLAT_DBG_LEVEL_WARNING 2
+#define OT_PLAT_DBG_LEVEL_INFO 3
+#define OT_PLAT_DBG_LEVEL_DEBUG 4
+
+#ifndef OT_PLAT_DBG_LVL
+#define OT_PLAT_DBG_LVL OT_PLAT_DBG_LEVEL_NONE
+#endif
+
+#if (OT_PLAT_DBG_LVL > OT_PLAT_DBG_LEVEL_NONE)
+/* if OT_STACK_ENABLE_LOG the log module will be already enabled */
+#ifndef OT_STACK_ENABLE_LOG
+#define LOG_ENABLE 1
+#define LOG_ENABLE_TIMESTAMP 1
+#include "fsl_component_log.h"
+#ifndef FREERTOS_LOG
+#define LOG_MODULE_NAME ot_plat_log
+LOG_MODULE_DEFINE(LOG_MODULE_NAME, kLOG_LevelDebug);
+#endif
+#endif
+#endif
+
+#if (OT_PLAT_DBG_LVL >= OT_PLAT_DBG_LEVEL_DEBUG)
+#define OT_PLAT_DBG(fmt, ...) LOG_DBG("%s " fmt, __func__, ##__VA_ARGS__)
+#define OT_PLAT_DBG_NO_FUNC(fmt, ...) LOG_DBG("%s " fmt, "", ##__VA_ARGS__)
+#else
+#define OT_PLAT_DBG(...)
+#define OT_PLAT_DBG_NO_FUNC(...)
+#endif
+#if (OT_PLAT_DBG_LVL >= OT_PLAT_DBG_LEVEL_INFO)
+#define OT_PLAT_INFO(fmt, ...) LOG_INF("%s " fmt, __func__, ##__VA_ARGS__)
+#else
+#define OT_PLAT_INFO(...)
+#endif
+#if (OT_PLAT_DBG_LVL >= OT_PLAT_DBG_LEVEL_WARNING)
+#define OT_PLAT_WARN(fmt, ...) LOG_WRN("%s " fmt, __func__, ##__VA_ARGS__)
+#else
+#define OT_PLAT_WARN(...)
+#endif
+#if (OT_PLAT_DBG_LVL >= OT_PLAT_DBG_LEVEL_ERR)
+#define OT_PLAT_ERR(fmt, ...) LOG_ERR("%s " fmt, __func__, ##__VA_ARGS__)
+#else
+#define OT_PLAT_ERR(...)
+#endif
+
+/**
+ * This structure represents different CCA mode configurations before Tx.
+ */
+typedef struct otCCAModeConfig_tag
+{
+    uint8_t
+        mCcaMode; ///< CCA Mode type [CCA1=0x01, CCA2=0x02, CCA3=0x03[CCA1 AND CCA2], CCA3=4[CCA1 OR CCA2], NoCCA=0xFF].
+    uint8_t mCca1Threshold;      ///< Energy threshold for CCA Mode1.
+    uint8_t mCca2CorrThreshold;  ///< CCA Mode 2 Correlation Threshold.
+    uint8_t mCca2MinNumOfCorrTh; ///< CCA Mode 2 Threshold Number of Correlation Peaks.
+} otCCAModeConfig;
+
+/**
+ * This function initializes the alarm service used by OpenThread.
+ *
+ */
+void otPlatAlarmInit(void);
+
+/**
+ * This function deinitialized the alarm service used by OpenThread.
+ *
+ */
+void otPlatAlarmDeinit(void);
+
+/**
+ * This function performs alarm driver processing.
+ *
+ * @param[in]  aInstance  The OpenThread instance structure.
+ *
+ */
+void otPlatAlarmProcess(otInstance *aInstance);
+
+/**
+ * This function initializes the radio service used by OpenThread.
+ *
+ */
+void otPlatRadioInit(void);
```

Comment:
So is this some kind of standardized API from OpenThead? Why copy the file then, we don't copy other platform headers from OT to Zephyr.